### ISSUE TITLE
Add in missing numerical info for items and entities

### ISSUE THREAD

**Issue by LevFendi** [Factorio-Access/FactorioAccess#135]
The description of an item usually includes the numerical info added in by hand. In other cases, reading an entity or checking its status reveals numerical info. Such info might be wondered without having to build a building for it.

To address numerical info in a unified way, it may be best to add a new info key that lists numerical specifications about a selected item (placed entity dimensions, power usage, working speed, module slot count, and so on).

**Comment by ahicks92** [comment]
I'd probably just fold this into the description: "builds things. Dimensions: 3x3, consumes 5 kilowatts"

We want people to discover it and are also running out of keys as well.  As long as it's after the end of the description, it shouldn't be annoying to stick it there, and I at least wouldn't consult it a ton on an ongoing basis.  Same with ] to check statuses, it can just go on the end there.  IMO it's fine to put information that you're going to look at 10 times in a whole run or something after the important stuff, because the slowdown is almost O(1) and minor in any case.

**Comment by LevFendi** [comment]
Alright yes, indeed. I will keep the static numerical info in the descriptions and the dynamic numerical info in the status checks. This post is a good place to build the full list of what numerical info exists, and which parts are not integrated yet.

**Comment by ahicks92** [comment]
What I know of:

- On items and entities, static electric info (not sure if the "consuming" at runtime on buildings is).
- Pollution production per object
- On pipes it'd be good to get fluid level without going into menus.
- Also fluid levels for storage tanks

I think there's others.  I am blanking on what they are.  Even that small list--especially the pipe ones--would be a lot of good QoL.

**Comment by LevFendi** [comment]
- I added fluid level info since it is dynamic. 
- Also I will later add current pollution output as a dynamic value as part of the status check.
- Meanwhile the static info will require manually updating all entity descriptions, which I will work on later. 
Here is the planned format for the revised descriptions: 
1. Building dimensions
2. Item/entity description
3. Building base power consumption OR fuel item fuel value
4. Building base pollution output

**Comment by ahicks92** [comment]
Dimensions are one of the least important things, should move that down.  Everything is squares, most things fall into buckets, you'll memorize it in an hour at most.
I'll throw out that while it is the case that most info is static, that info is found either on the prototype or, at runtime, on the entity.  Changing descriptions to include the static info will break mods which change or add entities, but reading the value gets us stuff "for free".

Probably that's occurred to you but if you're not me who reads API docs when bored possible you didn't know it's there.  I know it's obvious, I'm not trying to be patronizing, it's just important enough that I'll mention it.

**Comment by LevFendi** [comment]
Yes good point, i found the values for dimensions and max power use and pollution emmissions in entity prototypes.

This kind of info can be programatically  appended after the description for all entities that have it.

**Comment by LevFendi** [comment]
Note: this should also include armor equipment

**Comment by rashadnaqeeb** [comment]
Could turret range also be added to turret descriptions?

**Comment by ahicks92** [comment]
Yes; the mechanism I proposed on Discord literally supports adding anything you'd ever want to ask for.  If we do it that way this is idk, 5-10 lines.
